pkg/cloudprovider/providers/openstack/openstack_volumes.go

```diff
@@ -553,14 +552,10 @@ func (os *OpenStack) getDevicePathFromInstanceMetadata(volumeID string) string {
 func (os *OpenStack) GetDevicePath(volumeID string) string {
 	devicePath := os.GetDevicePathBySerialID(volumeID)
 
-	if devicePath == "" {
-		devicePath = os.getDevicePathFromInstanceMetadata(volumeID)
-	}
-
 	if devicePath == "" {
 		glog.Warningf("Failed to find device for the volumeID: %q", volumeID)
+		devicePath = os.getDevicePathFromInstanceMetadata(volumeID)
```

Comment:
We want to get device path from `id` and if necessary from `metadata service`, if no one succeeds, then log error message. I don't think we need to do this change.